File: downloader.py
#!/usr/bin/env python3

import requests


def fetch(url):
    dest = './files/' + ('/'.join(url.split('/')[3:]));
    r = requests.get(url)
    with open(dest, 'wb') as fd:
        for chunk in r.iter_content(chunk_size=1024):
            fd.write(chunk)
    print('END   [%d/%d] %s' % (current, total, url.strip()))


with open('./list.txt') as f:
    urls = f.readlines()

total = len(urls)
current = 0
print('Total: %d' % total)

for url in urls:
    url = url.strip()
    if url:
        current += 1
        print('START [%d/%d] %s' % (current, total, url.strip()))
        fetch(url)

File: downloader.js
var fs = require('fs');
var list = fs.readFileSync('list.txt').toString().split('\n');
var http = require('http');

var done = 0,
    total = list.length;

(function download(list) {
    var url = list.pop();

    if (url) {
        var dest = './files/' + url.split('/').slice(3).join('/'),
            file = fs.createWriteStream(dest);

        http.get(url, function(response) {
            response.pipe(file);
            file.on('finish', function () {
                file.close(function () {
                    console.log('[', ++done + '/' + total + '] ' + url);
                    if (list.length) {
                        download(list);
                    }
                })
            });
        }).on('error', function (e) {
            fs.unlink(dest);
            console.error(e);
        });
    } else if (list.length) {
        download(list);
    }
})(list);
